impl/src/main/java/org/ehcache/config/serializer/DefaultSerializationProviderConfiguration.java

```diff
@@ -41,37 +44,38 @@
     if (serializerClass == null) {
       throw new NullPointerException("Serializer class cannot be null");
     }
+    
     String alias = serializableClass.getName();
+    Constructor<? extends Serializer<T>> transientConstructor = null;
+    Constructor<? extends Serializer<T>> persistentConstructor = null;
     
-    if (serializerClass.isAnnotationPresent(Serializer.Transient.class)) {
```

Comment:
This set of changes would be simpler if you extracted an isConstructorPresent(...) method which could do the exception handling.